api/v1alpha1/shared_types.go

```diff
@@ -609,3 +609,35 @@ type HTTP2Settings struct {
 	// +optional
 	OnInvalidMessage *InvalidMessageAction `json:"onInvalidMessage,omitempty"`
 }
+
+// CustomErrorResponse defines the configuration to override specific responses with a custom one.
+type CustomErrorResponse struct {
+	// Match configuration.
+	Match CustomErrorResponseMatch `json:"match"`
+	// Response configuration.
+	Response CustomResponse `json:"response"`
```

Comment:
If we lean more strongly into the "override" terminology for this feature, we could change `response` to `overrides`. I would have to double check the xDS, but I believe you can do a "partial override" to do things like change/add response headers, but only when an error happens. I was asked the other day if Emissary/Edge Stack supported that for a large org that needed it and while Emissary/Edge Stack currently don't it would be cool to support it here if possible.